amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/Lease.java

```diff
@@ -171,6 +171,15 @@ public void update(final Lease lease) {
         childShardIds(lease.childShardIds);
     }
 
+    /**
+     * @param leaseDurationNanos duration of lease in nanoseconds
+     * @param asOfNanos time in nanoseconds to check expiration as-of
+     * @return true if lease lease is ready te taken
+     */
+    public boolean isAvailable(long leaseDurationNanos, long asOfNanos) {
```

Comment:
Prefer a [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) type instead of `long nanos`.

Reply:
Because we have this all the way up I'm not going to change it right now.